ext/zip/php_zip.c

```diff
@@ -2097,6 +2097,40 @@ PHP_METHOD(ZipArchive, getArchiveComment)
 }
 /* }}} */
 
+PHP_METHOD(ZipArchive, setArchiveFlag)
+{
+	struct zip *intern;
+	zval *self = ZEND_THIS;
+	zend_long flag, value;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "ll", &flag, &value) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	ZIP_FROM_OBJECT(intern, self);
+
+	if (zip_set_archive_flag(intern, flag, (int)value)) {
```

Comment:
It might make sense to check that the value provided fits within an ``int`` and throw a ValueError if not, I know we do this in some other cases/extensions.

Reply:
this value is rather an enum (a value among set of constants, ZIP_AFL_*)
so if we want a ValueError we should check supported values
but we can't know how this will change in future version of the library

Also libzip will return a error on bad value, so IMHO it will be strange to have 3 kinds of error (1 for php, 1 for libzip), so I prefer to not raise exception here

Reply:
ACK